Source/GoBot/GoBot.swift

```diff
@@ -1397,13 +1397,13 @@ class GoBot: Bot {
     func markMessageAsRead(_ message: MessageIdentifier) {
         userInitiatedQueue.async {
             do {
-                let report = try self.database.report(for: message)
+                let isRead = try self.database.isReportRead(for: message)
                 try self.database.markMessageAsRead(identifier: message, isRead: true)
-                if let report = report, report.isUnread {
+                if !isRead {
                     NotificationCenter.default.post(
```

Comment:
Something is going wrong here but I'm not sure what it is. As I scroll through the home feed I see a ton of `CountUnreadNotificationsOperation`s being triggered and I think they are being triggered by this notification. So I put a breakpoint in here and saw that this notification is being posted more than once for a given message. So like if I scroll past a message once, the code above runs to mark it as read, but then if I scroll past it again `self.database.isReportRead(for: message)` returns `false` and it tries to mark it as read again.

Reply:
I pushed another commit that fixes that. The issue was that that code was not making distinction between the messages in the home feed (just a few have reports associated with them) and messages in the notifications screen (all of them have associated reports)